lib/vanagon/platform/windows.rb

```diff
@@ -174,22 +187,17 @@ def generate_msi_package(project)
           "mkdir -p $(tempdir)/{SourceDir,wix/wixobj}",
           "#{@copy} -r wix/* $(tempdir)/wix/",
           "gunzip -c #{project.name}-#{project.version}.tar.gz | '#{@tar}' -C '$(tempdir)/SourceDir' --strip-components 1 -xf -",
-          # Run the Heat command in a single pass
-          # Heat command documentation at: http://wixtoolset.org/documentation/manual/v3/overview/heat.html
-          #   dir <directory> - Traverse directory to find all sub-files and directories.
-          #   -ke             - Keep Empty directories
-          #   -cg             - Component Group Name
-          #   -gg             - Generate GUIDS now
-          #   -srd            - Suppress root element generation, we want to reference the default root element
-          #                     TARGETDIR in the project.wxs file, not a newly generated one.
-          #   -sreg           - Suppress registry harvesting.
-          "cd $(tempdir); \"$$WIX/bin/heat.exe\" dir SourceDir -v -ke -indent 2 -cg #{cg_name} -gg -srd -t wix/filter.xslt -sreg -out wix/#{project.name}-harvest.wxs",
+          "cd $(tempdir); \"$$WIX/bin/heat.exe\" dir #{application_source_path} #{application_heat_flags} -out wix/#{project.name}-harvest-app.wxs",
+          "cd $(tempdir); \"$$WIX/bin/heat.exe\" dir #{appdata_source_path} #{appdata_heat_flags}  -out wix/#{project.name}-harvest-programdata.wxs",
           # Apply Candle command to all *.wxs files - generates .wixobj files in wix directory.
           # cygpath conversion is necessary as candle is unable to handle posix path specs
-          "cd $(tempdir)/wix/wixobj; for wix_file in `find $(tempdir)/wix -name \'*.wxs\'`; do \"$$WIX/bin/candle.exe\" #{candle_flags} $$(cygpath -aw $$wix_file) || exit 1; done",
+          # the preprocessor variables AppDataSourcePath and ApplicationSourcePath are required due to the -var input to the heat
+          # runs listed above.
+          "cd $(tempdir)/wix/wixobj; for wix_file in `find $(tempdir)/wix -name \'*.wxs\'`; do \"$$WIX/bin/candle.exe\" #{candle_flags} -dApplicationSourcePath=\"#{application_source_path}\" -dAppDataSourcePath=\"#{appdata_source_path}\" $$(cygpath -aw $$wix_file) || exit 1; done",
           # run all wix objects through light to produce the msi
           # the -b flag simply points light to where the SourceDir location is
-          "cd $(tempdir)/wix/wixobj; \"$$WIX/bin/light.exe\" #{light_flags} -b $$(cygpath -aw $(tempdir)) -out $$(cygpath -aw $(workdir)/output/#{target_dir}/#{msi_package_name(project)}) *.wixobj",
+          # -loc is required for the UI localization it points to the actual localization .wxl
+          "cd $(tempdir)/wix/wixobj; \"$$WIX/bin/light.exe\" #{light_flags} -b $$(cygpath -aw $(tempdir)) -loc $$(cygpath -aw $(tempdir)/wix/localization/puppet_en-us.wxl) -out $$(cygpath -aw $(workdir)/output/#{target_dir}/#{msi_package_name(project)}) *.wixobj",
```

Comment:
The ```-loc $$(cygpath -aw $(tempdir)/wix/localization/puppet_en-us.wxl)``` belongs to the UI changes - would prefer if you didn't include it in this change as I have a specific comment related to it in the UI work.